### PR TITLE
persist: Delete `arrow2` and 'persist_use_arrow_rs_library' dyncfg

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR removes `arrow2` from our dependency tree entirely, and the feature gate that enabled switching between `arrow2` and `arrow-rs`.

Note: I'm not expecting any issues, but to be on the safe side I would like to keep our `arrow2` fork around for a few weeks, just in case we need to revert this PR.

### Motivation

We already migrated all of our stats collection to `arrow-rs` and today I moved the feature flag `persist_use_arrow_rs_library` to 100% of users. Removing `arrow2` simplifies our code and makes it easier to rebase https://github.com/MaterializeInc/materialize/pull/26561, introducing structured data into Persist.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

**Review by bkirwi — APPROVED** [review]
Clean cut!

File: src/persist/src/indexed/columnar.rs [hunk 1]
```diff
@@ -283,11 +289,26 @@ impl<'a> ColumnarRecordsRef<'a> {
         if idx >= self.len {
             return None;
         }
+
+        let key_start = self.key_offsets[idx]
```
Comment (danhhz) [inline src/persist/src/indexed/columnar.rs:293]: Ooc what type is it? We could potentially also use CastFrom here

**Review by danhhz — APPROVED** [review]
Yessssssss \o/

File: src/persist/src/indexed/columnar.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/persist/src/indexed/columnar.rs:293]: I was hoping we could do the same, unfortunately it's an `i32` :/

**Pull request merged**